pgserver/logrepl/subscription.go

```diff
@@ -1,35 +1,136 @@
 package logrepl
 
 import (
-	"context"
 	stdsql "database/sql"
-
+	"errors"
+	"fmt"
 	"github.com/apecloud/myduckserver/adapter"
 	"github.com/apecloud/myduckserver/catalog"
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/jackc/pglogrepl"
+	"sync"
 )
 
-func WriteSubscription(ctx *sql.Context, name, conn, pub string) error {
-	_, err := adapter.ExecCatalogInTxn(ctx, catalog.InternalTables.PgSubscription.UpsertStmt(), name, conn, pub)
-	return err
+type Subscription struct {
+	Subscription string
+	Conn         string
+	Publication  string
+	LsnStr       string
+	Enabled      bool
+	Replicator   *LogicalReplicator
 }
 
-func FindReplication(db *stdsql.DB) (name, conn, pub string, ok bool, err error) {
-	var rows *stdsql.Rows
-	rows, err = db.QueryContext(context.Background(), catalog.InternalTables.PgSubscription.SelectAllStmt())
+var keyColumns = []string{"subname"}
+var statusValueColumns = []string{"subenabled"}
+var lsnValueColumns = []string{"subskiplsn"}
+
+var subscriptionMap = sync.Map{}
+var mu sync.Mutex
+
+func UpdateSubscriptions(ctx *sql.Context) error {
+	mu.Lock()
+	defer mu.Unlock()
+	rows, err := adapter.QueryCatalog(ctx, catalog.InternalTables.PgSubscription.SelectAllStmt())
 	if err != nil {
-		return
+		return err
 	}
 	defer rows.Close()
 
-	if !rows.Next() {
-		return
+	var tempMap = make(map[string]*Subscription)
```

Comment:
The `temp` prefix is unnecessary. It is a local variable, implying it is temporary.